datalake/networking_stack.py

```diff
@@ -32,5 +32,6 @@ def __init__(self, scope: core.Construct, stack_id: str, **kwargs) -> None:
                     reserved=True,
                 ),
             ],
+            max_azs=99 if deploy_env == "prod" else 1,
```

Comment:
We should pull out an [enum](https://docs.python.org/3/library/enum.html) for the environment names sooner rather than later. Up to you whether you want to create a separate PR for that.

Reply:
I would rather add enum in other PR. It needs to be used at multiple places.

Reply:
> Also, I expect max_azs=None is more appropriate than 99.

From CDK docs describing `max_azs` parameter looks like using value such as `99` is recommended.

"""
To use “all AZs” available to your account, use a high number (such as 99).
"""

Reply:
Docs is not saying what's going to happen if I set None.